src/prng/hc128.rs

```diff
@@ -129,15 +133,12 @@ impl Hc128Rng {
         let mut state = Hc128Rng {
             state: Hc128 { t: t, counter1024: 0 },
             results: [0; 16],
-            index: 0,
+            index: 16, // generate on first use
         };
 
         // run the cipher 1024 steps
         for _ in 0..64 { state.state.sixteen_steps() };
         state.state.counter1024 = 0;
-
-        // Prepare the first set of results
-        state.state.update(&mut state.results);
```

Comment:
Why did this code get dropped?

I also noticed a similar difference with ISAAC when merging upstream into my branch; not sure why it was there.

Isn't it better to do as much as possible during init?

Reply:
To be honest, I don't completely remember. I had reasons for it when working on the `BlockRngCore` stuff, this change to the seeding was part of the 'functional' changes.

ChaCha generated the set of results on first use, while ISAAC and HC-128 did on initialization. So It seemed a good idea to pick one default.

Generating results on first use made the seeding function for slightly simpler, it is just initializing a struct:
```rust
impl<R> SeedableRng for BlockRng<R>
where R: BlockRngCore + SeedableRng,
{
    type Seed = R::Seed;

    fn from_seed(seed: Self::Seed) -> Self {
        let results_empty = R::Results::default();
        Self {
            core: R::from_seed(seed),
            index: results_empty.as_ref().len(), // generate on first use
            results: results_empty,
        }
    }

    fn from_rng<Rseed: Rng>(rng: Rseed) -> Result<Self, Error> {
        let results_empty = R::Results::default();
        Ok(Self {
            core: R::from_rng(rng)?,
            index: results_empty.as_ref().len(), // generate on first use
            results: results_empty,
        })
    }
}
```

I feel like I had a better reason, but can't remember it.